x-pack/plugins/security_solution/public/detections/components/rules/rule_preview/preview_logs.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Fragment, useMemo } from 'react';
+import { EuiCallOut, EuiText, EuiSpacer, EuiAccordion } from '@elastic/eui';
+import { RulePreviewLogs } from '../../../../../common/detection_engine/schemas/request';
+import * as i18n from './translations';
+
+interface PreviewLogsComponentProps {
+  logs: RulePreviewLogs[];
+  hasNoiseWarning: boolean;
+}
+
+interface SortedLogs {
+  startedAt?: string;
+  logs: string[];
+}
+
+interface LogAccordionProps {
+  logs: SortedLogs[];
+  isError?: boolean;
+}
+
+export const PreviewLogsComponent: React.FC<PreviewLogsComponentProps> = ({
+  logs,
+  hasNoiseWarning,
+}) => {
+  const sortedLogs = useMemo(
+    () =>
+      logs.reduce<{
+        errors: SortedLogs[];
+        warnings: SortedLogs[];
+      }>(
+        (acc, curr) => ({
+          errors:
+            curr.errors.length > 0 // Filter out empty error logs
```

Comment:
we could probably define a tiny function outside the scope of this function called `addLogs` that looks something like `(startedAt, logs, allLogs) => logs.length ? [{startedAt, logs}, ...allLogs] : allLogs`, and then the reducer can get simplified like `({errors, warnings}, curr) => ({errors: addLogs(curr.startedAt, curr.errors, errors), warnings: addLogs(curr.startedAt, curr.warnings, warnings)})`